modules/codegen/src/main/scala/com/twilio/guardrail/generators/CirceProtocolGenerator.scala

```diff
@@ -173,7 +174,13 @@ object CirceProtocolGenerator {
         val params = (parents.reverse.flatMap(_.params) ++ selfParams).filterNot(
           param => discriminatorNames.contains(param.term.name.value)
         )
-        val terms = params.map(_.term)
+
+        import ScalaLanguage._
+
+        val deduplicatedParams = params.foldLeft(List.empty[ProtocolParameter[ScalaLanguage]]) { (s, a) =>
+          s.find(p => Eq.eqv(a, p)).map(_ => s).getOrElse(a :: s)
+        }
+        val terms = deduplicatedParams.map(_.term).reverse // not needed but without it we break bunch of tests
```

Comment:
Is it possible to do this just in ProtocolGenerator rather than needing to do this for each concrete impl?  JacksonGenerator will need this too.

Also I think we should just bail with an error if there are two params with the same name and they aren't "compatible".

Reply:
well it would be possible to move this logic to ProtocolGenerator, but this would require larger refactoring - as of now, there is already a lot of specific logic regarding how extensions and parents are handled - i don't feel qualified for that refactoring

do you feel that this is required before for merge to be considered? if so, i can take a stab at it but honestly i would rather just solve my own problem

regarding the error, that is a fair point

Reply:
That's fair.  I've been eyeing some of the redundant protocolgenerator code for a refactor for a while now, so I might look into it after this PR lands.  No need to tackle that before getting this in, though.

Reply:
ok, i will attack your other concern then